pkg/operator/ceph/cluster/controller.go

```diff
@@ -451,31 +456,34 @@ func (c *ClusterController) onUpdate(oldObj, newObj interface{}) {
 		return
 	}
 
+	cluster.status.SetMessage("updating cluster...")
 	err = wait.Poll(updateClusterInterval, updateClusterTimeout, func() (bool, error) {
 		return c.handleUpdate(newClust.Name, cluster)
 	})
 	if err != nil {
 		message := fmt.Sprintf("giving up trying to update cluster in namespace %s after %s", cluster.Namespace, updateClusterTimeout)
 		logger.Error(message)
-		if err := c.updateClusterStatus(newClust.Namespace, newClust.Name, cephv1.ClusterStateError, message); err != nil {
+		if err := c.updateClusterStatus(cluster, newClust.Name, cephv1.ClusterStateError, message); err != nil {
 			logger.Errorf("failed to update cluster status in namespace %s: %+v", newClust.Namespace, err)
 		}
 		return
 	}
 }
 
 func (c *ClusterController) handleUpdate(crdName string, cluster *cluster) (bool, error) {
-	if err := c.updateClusterStatus(cluster.Namespace, crdName, cephv1.ClusterStateUpdating, ""); err != nil {
+	if err := c.updateClusterStatus(cluster, crdName, cephv1.ClusterStateUpdating, ""); err != nil {
 		logger.Errorf("failed to update cluster status in namespace %s: %+v", cluster.Namespace, err)
 		return false, nil
 	}
 
-	if err := cluster.createInstance(c.rookImage, cluster.Info.CephVersion); err != nil {
-		logger.Errorf("failed to update cluster in namespace %s. %+v", cluster.Namespace, err)
+	if err := cluster.createInstance(c.rookImage); err != nil {
+		errMsg := "failed to update cluster in namespace %s. %+v"
+		cluster.status.SetMessage(fmt.Sprintf(errMsg, cluster.Namespace, err))
```

Comment:
If `updateClusterStatus()` isn't called, setting the message on the `cluster.status` won't have any effect, right? This status will be applied elsewhere, right?

Reply:
true, it is applied in [line 486](https://github.com/rook/rook/pull/2638/files#diff-8a603243c4e06abaf58c1cbd1426cf25R486)